### PR TITLE
Update to golang 1.23

### PR DESCRIPTION
**Pull request by yob**
In #61 we added a dependabot config to this project. However, dependabot is failing with an error:

    Dependabot can't resolve your Go dependency files
    Dependabot failed to update your dependencies because there was an error resolving your Go dependency files.
    Dependabot encountered the following error:
    maps: package maps is not in GOROOT (/home/dependabot/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.20.10.linux-amd64/src/maps)

It's not super clear to me what's causing that, but I notice it's talking about go 1.20, when this relative old and stable project is still setup to build with go 1.16.

I thought I'd give it a little bump. Certainly can't hurt.

I also snuck in dependabot config to occasionally bump the golang version used. Hopefully that stops it falling so far behind again

### CODE REVIEW

**Review by wolfeidau — APPROVED** [review]
👍🏻 🚀

**Comment by yob** [comment]
I'll hold off merging until #63 lands, so I can see the tests for this PR go green

**Pull request merged**
